### PR TITLE
Adding Rainbowkit + Wagmi Explanation and Gnosis Chain setup

### PR DESCRIPTION
**Pull request by billyjitsu**
## Adding Guide to "Interacting with Gnosis"

- Created a new section called "Using Rainbow Kit + Wagmi" that explains how to setup Rainbow Kit/Wagmi to accept Gnosis Chain as a custom chain

Adding sidebar.js link,  Interact/Readme link, and adding Rainbowkit+Wagmi guide within structure

### CODE REVIEW

File: docs/developers/interact/rainbowkit-wagmi.mdx [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+On the top of our _app.js file, we import jsonRpcProvider
+
+```js showLineNumbers
+import { alchemyProvider } from "wagmi/providers/alchemy";
+import { publicProvider } from "wagmi/providers/public";
+import { jsonRpcProvider } from 'wagmi/providers/jsonRpc'; //<<< new RPC
+
+```
+
+Now that we imported it, we can add it to our "configureChains" hook and set it up. You will see the "jsonRpcProvider" is on the top of alchemy and public. **It is manually setup to use the Gnosis Chain RPC (using ANKR for this example, but can use Quicknode or Pokt)**, but want to show that you can use ANY RPC of your liking for this. You are not limited to the preset RPCs that wagmi supports. So now the order of failure would be , ankr, alchemy and then public RPCs (Note that Alchemy **does NOT currently** support Gnosis Chain, just using it as an example to help you understand the concept):
```
Comment (alebanzas) [inline docs/developers/interact/rainbowkit-wagmi.mdx:269]: I will add here a link to https://docs.gnosischain.com/tools/rpc/

**Review by alebanzas — CHANGES_REQUESTED** [review]
Thanks @billyjitsu ! I would only add some links to the current RPC providers to make the guide "provider agnostic".

File: docs/developers/interact/rainbowkit-wagmi.mdx (hunk 1 above)
Comment (billyjitsu) [reply docs/developers/interact/rainbowkit-wagmi.mdx]: Pushed with changes

**Review by alebanzas — APPROVED** [review]
lgtm, thanks @billyjitsu !

**Pull request merged**